app/Support/EmailOutboxSchema.php

```
@@ -0,0 +1,73 @@
+                LIMIT 1
+            ");
+            if ($exists && $exists->num_rows > 0) {
+                return self::$availability[$connectionId] = true;
+            }
+
+            // Never let compatibility DDL implicitly commit a circulation
+            // transaction. Pre-migration callers simply send best-effort; the
+            // installer/updater creates the table authoritatively.
+            $transaction = $db->query('SELECT @@session.in_transaction AS active')->fetch_assoc();
+            if ((int) ($transaction['active'] ?? 0) === 1) {
+                SecureLogger::warning('Email outbox creation deferred until outside the active transaction');
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'App\Support\SecureLogger' in method 'ensure'. (undefined)

(StaticAccess)